PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/SparkServerlessClusterOpsCtrl.java

```diff
@@ -38,38 +39,24 @@ public SparkServerlessClusterOpsCtrl(@NotNull SparkServerlessClusterOps sparkSer
         this.sparkServerlessClusterOps = sparkServerlessClusterOps;
 
         this.sparkServerlessClusterOps.getDestroyAction().subscribe(triplet -> {
-            LOG.info(String.format("Message received. AdlAccount: %s, clusterName: %s, currentNode: %s",
+            LOG.info(String.format("Destroy message received. AdlAccount: %s, clusterName: %s, currentNode: %s",
                     triplet.getLeft(), triplet.getMiddle(), triplet.getRight()));
 
-            // TODO: pop up a destroy dialog
-
-            // TODO: Update confirmDestroyAction
-            boolean confirmDestroyAction = true;
-            if (confirmDestroyAction) {
-                // instruct parent node to remove this node
-                DefaultLoader.getIdeHelper().invokeLater(() -> {
-                    Node currentNode = triplet.getRight();
-                    currentNode.getParent().removeDirectChildNode(currentNode);
-                });
-            }
+            DefaultLoader.getIdeHelper().invokeLater(() -> {
+                SparkServerlessClusterDestoryDialog destroyDialog = new SparkServerlessClusterDestoryDialog(
```

Comment:
Try `IdeSchedulers` with this, refer to `AddNewClusterCtrlProvider` please, rather than `DefaultLoader.getIdeHelper().invokeLater()`